ngraph/frontend/onnx_editor/include/onnx_editor/edge_mapper.hpp

```diff
@@ -0,0 +1,98 @@
+            ///        with the input name (providing an input index is not enough).
+            ///        If a unique edge is found, it will be returned.
+            ///        If InputEdge cannot be determined based on parameter values an ngraph_error
+            ///        exception will be thrown.
+            ///
+            /// \param node An EditorNode helper structure created based on a node name
+            ///             or a node output name.
+            ///
+            /// \param input An EditorInput helper structure created based on a input name
+            ///              or a input index.
+            InputEdge find_input_edge(const EditorNode& node, const EditorInput& input) const;
+
+            /// \brief Returns an OutputEdge based on a node (node name or output name)
+            ///        and an output (output name or output index).
+            ///
+            /// \note  The node name can be ambiguous (many ONNX nodes can have the same name).
+            ///        In such a case the algorthim will try to match the given node name
+            ///        with the output name (providing an output index is not enough).
+            ///        If after such operation a found edge is unique, it is returned.
+            ///        If OutputEdge cannot be determined based on given params the ngraph_error
+            ///        exception is thrown.
+            ///
+            /// \param node An EditorNode helper structure created based on a node name
+            ///             or a node output name.
+            ///
+            /// \param output An EditorOutput helper structure created based on a output name
+            ///               or a output index.
+            OutputEdge find_output_edge(const EditorNode& node, const EditorOutput& output) const;
+
+            /// \brief Returns an OutputEdge based on a output name.
+            ///
+            /// \note  The output name guarantees the uniqueness of the edge.
+            ///
+            /// \param output_name A node output name.
+            ///
+            OutputEdge find_output_edge(const std::string& output_name) const;
+
+            /// \brief Updates state of a EdgeMapper instance if a model was changed.
+            ///
+            /// \param graph_proto Reference to a GraphProto object.
+            ///
+            void update(const ONNX_NAMESPACE::GraphProto& graph_proto);
+
+        private:
+            std::vector<int> find_node_indexes(const std::string& node_name,
+                                               const std::string& output_name) const;
+            std::string get_node_input_name(int node_index, int input_index) const;
+            std::string get_node_output_name(int node_index, int output_index) const;
+
+            std::vector<std::vector<std::string>> m_node_inputs;
+            std::vector<std::vector<std::string>> m_node_outputs;
+            std::multimap<std::string, int> m_node_name_to_index;
```

Comment:
Is it really a `node_name`? In one of the methods, `find_node_indexes`, it is used to map output name, so tensor name, not a node name.

Reply:
To be more clear, it is the map of node names and output name:
For example, if we have a node:
```
  node {
    input: "in1"
    output: "relu1"
    op_type: "Relu"
    name: "my_relu"
  }
```
in `m_node_name_to_index` we have:
```
{"reulu1", 0},
{"my_relu", 0}
```

Reply:
But tensor name space and node name space can overlap. If it really happens, a given name may be mapped to two different node indices, and we cannot say which one is what we need without specifying namespace for a given name.

Reply:
right, fixed in https://github.com/openvinotoolkit/openvino/pull/4927/commits/641d6a9d0bb9d7ebb5dad79d8119f4145a6ef162
Now, If there is a collision between `output_name` and `node_name` higher priority has `output_name` (as sth unambiguous). Is it ok for you?